bpf/lib/nodeport.h

```diff
@@ -449,6 +450,37 @@ static __always_inline int dsr_reply_icmp6(struct __ctx_buff *ctx,
 		.daddr		= ip6->saddr,
 		.payload_len	= bpf_htons((__u16)payload_len),
 	};
+	struct ipv6hdr inner_ipv6_hdr __align_stack_8 = *ip6;
+	__be16 dport;
+	__s32 l4_dport_offset;
+
+	/*
+	 * DSR changes the destination address from service ip to pod ip and
+	 * destination port from service port to pod port. While resppnding
+	 * back with ICMP error, it is necessary to set it to original ip and
+	 * port.
+	 */
+	union v6addr addr = {
+		.p1 = ctx_load_meta(ctx, CB_ADDR_V6_1),
+		.p2 = ctx_load_meta(ctx, CB_ADDR_V6_2),
+		.p3 = ctx_load_meta(ctx, CB_ADDR_V6_3),
+		.p4 = ctx_load_meta(ctx, CB_ADDR_V6_4),
+	};
```

Comment:
I think we shouldn't rely on the metadata here. It can be accidentally broken by someone outside of this function. Instead, we should pass `&addr` here.

https://github.com/cilium/cilium/blob/04ecee7fb453a8b480ed9ba2b0f0a2389aeeda26/bpf/lib/nodeport.h#L573